### PR TITLE
fix #3116

### LINKED ISSUE #3116: Builder frequently has full inventory when going to restaurant.
### Minecolonies version
Version: 0.10.27 - 0.10.54

### Expected behavior
Builder Tries to Dump inventory in Build Hut first then goes to stuff his/her face at restaurant. (Or at least tries to free one spot up for food)

### Actual behaviour
Builder heads straight to restaurant with a full inventory.

### Additional Notes
Have "builderInfiniteResources" set to true.
(Also Recommend having Builder zero his personal inventory of everything except food for the following situations [while builderInfiniteResources is set to true of course]: Finishing Construction, When construction is canceled, Going to Home/Townhall for the night, Going to the restaurant. Instead of dumping whatever is spawned into builder hut's inventory.

### Steps to reproduce the problem
1. Have Builder with Worker
2. Have stocked and staffed restuarant.
3. Have Builder build stuff.
4. Wait.

<bountysource-plugin>

---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/66787742-builder-frequently-has-full-inventory-when-going-to-restaurant?utm_campaign=plugin&utm_content=tracker%2F43175463&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F43175463&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by Raycoms**
Closes #3116

# Changes proposed in this pull request:
- Feed citizen on full inventory
-
-

Review please

### CODE REVIEW

**Comment by reggaemuffin** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/main/java/com/minecolonies/coremod/entity/ai/citizen/cook/EntityAIWorkCook.java  3
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Minecolonies/minecolonies/pullRequest?prid=5271819&bid=17267545)

File: src/main/java/com/minecolonies/coremod/entity/ai/citizen/cook/EntityAIWorkCook.java [hunk 1]
```diff
@@ -174,6 +174,19 @@ private IAIState serveFoodToCitizen()
 
         if (InventoryUtils.isItemHandlerFull(handler))
         {
+            if (!citizenToServe.isEmpty())
+            {
+                final int foodSlot = InventoryUtils.findFirstSlotInItemHandlerWith(worker.getInventoryCitizen(), ItemStackUtils.CAN_EAT);
+                if (foodSlot != -1)
+                {
+                    final ItemStack stack = worker.getInventoryCitizen().extractItem(foodSlot, 1, false);
```
Comment (marchermans) [inline src/main/java/com/minecolonies/coremod/entity/ai/citizen/cook/EntityAIWorkCook.java:183]: Please add in a safety check that checks if serving the first citizen is actually usefull.

**Review by marchermans — CHANGES_REQUESTED** [review]

File: src/main/java/com/minecolonies/coremod/entity/ai/citizen/cook/EntityAIWorkCook.java (hunk 1 above)
Comment (Raycoms) [reply src/main/java/com/minecolonies/coremod/entity/ai/citizen/cook/EntityAIWorkCook.java:183]: What do you mean?

Comment (marchermans) [reply src/main/java/com/minecolonies/coremod/entity/ai/citizen/cook/EntityAIWorkCook.java:183]: I mean, please check if it is useful to feed the first citizen
Or if it is more useful to feed another citizen with that piece of meal.

**Comment by reggaemuffin** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/main/java/com/minecolonies/coremod/entity/ai/citizen/cook/EntityAIWorkCook.java  3
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Minecolonies/minecolonies/pullRequest?prid=5271819&bid=17267545)

**Review by marchermans — APPROVED** [review]

**Comment by marvin-bitterlich** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/main/java/com/minecolonies/coremod/entity/ai/citizen/cook/EntityAIWorkCook.java  3
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Minecolonies/minecolonies/pullRequest?prid=5271819&bid=17267545)

**Pull request merged**